internal/service/userService.go

```diff
@@ -0,0 +1,69 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/repository"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/service/models"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/service/models/exceptions"
+	logger "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/topology"
+)
+
+type UserService interface {
+	SaveUser(ctx context.Context, user models.User) (*models.User, error)
+	GetUserById(ctx context.Context, id string) (*models.User, error)
+}
+
+type service struct {
+	userRepository repository.UserRepository
+}
+
+func NewUserService(repository repository.UserRepository) UserService {
+	return &service{
+		userRepository: repository,
+	}
+}
+
+const errorMessage = "something went wrong"
```

Comment:
You could have used [Sentinel Errors](https://medium.com/gopher-time/writing-clean-code-in-go-sentinel-errors-5ad93a30bc8e) for this and error wrapping for your error handling logic as opposed to your exception approach. Although many languages support this you won't see this kind of implementation on golang projects